From 356da128e2bf51344194f603bda3b0515fe630de Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Sun, 7 Apr 2024 12:26:35 +0800
Subject: [PATCH] overlay: SystemUI: Follow light/dark theme in power menu

Now that we've modified the power menu to refresh on UI mode changes
and use Monet colors, make it follow the system light/dark theme for
better integration in light mode.

Requires frameworks/base commit:
    SystemUI: Always refresh global actions dialog on UI mode change

Change-Id: Id6c97d5db16e494442494bf5dc1926a854c11ee7
---
 .../SystemUI/res/values-night/colors.xml      | 22 +++++++++++++++++++
 .../packages/SystemUI/res/values/colors.xml   | 22 +++++++++++++++++++
 2 files changed, 44 insertions(+)
 create mode 100644 overlay/common/frameworks/base/packages/SystemUI/res/values-night/colors.xml
 create mode 100644 overlay/common/frameworks/base/packages/SystemUI/res/values/colors.xml

diff --git a/overlay/common/frameworks/base/packages/SystemUI/res/values-night/colors.xml b/overlay/common/frameworks/base/packages/SystemUI/res/values-night/colors.xml
new file mode 100644
index 0000000000..e68e2cb93b
--- /dev/null
+++ b/overlay/common/frameworks/base/packages/SystemUI/res/values-night/colors.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    Copyright (C) 2021 The Proton AOSP Project
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+        http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<resources>
+    <!-- Set colors for power menu -->
+    <color name="global_actions_lite_background">@*android:color/primary_device_default_dark</color>
+    <color name="global_actions_lite_button_background">@*android:color/surface_dark</color>
+    <color name="global_actions_lite_text">@*android:color/foreground_device_default_dark</color>
+
+    <color name="global_actions_lite_emergency_background">@*android:color/surface_dark</color>
+    <color name="global_actions_lite_emergency_icon">@*android:color/accent_device_default_dark</color>
+</resources>
diff --git a/overlay/common/frameworks/base/packages/SystemUI/res/values/colors.xml b/overlay/common/frameworks/base/packages/SystemUI/res/values/colors.xml
new file mode 100644
index 0000000000..f25b30829d
--- /dev/null
+++ b/overlay/common/frameworks/base/packages/SystemUI/res/values/colors.xml
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+    Copyright (C) 2020 The Proton AOSP Project
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+        http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+-->
+<resources>
+    <!-- Set colors for power menu -->
+    <color name="global_actions_lite_background">@*android:color/primary_device_default_light</color>
+    <color name="global_actions_lite_button_background">@*android:color/surface_light</color>
+    <color name="global_actions_lite_text">@*android:color/foreground_device_default_light</color>
+
+    <color name="global_actions_lite_emergency_background">@*android:color/surface_light</color>
+    <color name="global_actions_lite_emergency_icon">@*android:color/accent_device_default_light</color>
+</resources>
